apiserver/facades/client/machinemanager/machinemanager.go

```diff
@@ -317,7 +323,7 @@ func (mm *MachineManagerAPI) addOneMachine(p params.AddMachineParams) (*state.Ma
 
 // ProvisioningScript returns a shell script that, when run,
 // provisions a machine agent on the machine executing the script.
-func (mm *MachineManagerAPI) ProvisioningScript(ctx context.Context, args params.ProvisioningScriptParams) (params.ProvisioningScriptResult, error) {
+func (mm *MachineManagerAPI) ProvisioningScript(ctx context.Context, controllerConfigGetter ControllerConfigGetter, args params.ProvisioningScriptParams) (params.ProvisioningScriptResult, error) {
```

Comment:
This is wrong, you can't pass this here. This should be on the MachingManagerAPI type.

Reply:
Got it. I used it as part of `MachineManagerAPI`, but then I figured out that we use this function directly in unit tests, and I used contollerConfigGetter as a parameter as one of the exits to pass the mocked controllerConfigService in the unit tests.

I'll think of how to sort it out in another way.

Reply:
fixed